engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -0,0 +1,314 @@
+		if err != nil {
+			// Retry strategy if something fails
+			logrus.Errorf("An error occurred creating a new idle enclave. Error\n%v", err)
+			fails++
+			if fails >= maxRetryAfterFails {
+				logrus.Debugf("The enclave pool sub-routine has been canceled after reaching the max retries value '%v'", maxRetryAfterFails)
+				break
+			}
+			logrus.Debugf("Next retry will be executed on '%v' seconds", timeBetweenRetries.Seconds())
+			time.Sleep(timeBetweenRetries)
+		}
+		enclaveUUID := enclave.EnclaveUUID(newEnclaveInfo.EnclaveUuid)
+
+		pool.idleEnclavesChan <- enclaveUUID
+		logrus.Debugf("Enclave with UUID '%v' was added intho the pool channel", enclaveUUID)
+	}
+}
+
+func (pool *EnclavePool) createNewEnclave() (*kurtosis_engine_rpc_api_bindings.EnclaveInfo, error) {
+	ctx := context.Background()
+
+	enclaveName, err := GetRandomIdleEnclaveName()
+	if err != nil {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred generating a random name for a new idle enclave.",
+		)
+	}
+
+	apiContainerVersion := pool.engineVersion
+
+	newEnclaveInfo, err := pool.enclaveCreator.CreateEnclave(
+		ctx,
+		apiContainerVersion,
+		defaultApiContainerLogLevel,
+		enclaveName,
+		defaultIsPartitioningEnabled,
+	)
+	if err != nil {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred creating idle enclave with name '%s', api container version '%v' and container log level '%s'",
+			enclaveName,
+			apiContainerVersion,
+			defaultApiContainerLogLevel,
+		)
+	}
+
+	logrus.Debugf("New idle enclave created '%+v'", newEnclaveInfo)
+	return newEnclaveInfo, nil
+}
+
+func (pool *EnclavePool) getRunningEnclave(ctx context.Context, enclaveUUID enclave.EnclaveUUID) (*enclave.Enclave, error) {
+	filters := &enclave.EnclaveFilters{
+		UUIDs: map[enclave.EnclaveUUID]bool{
+			enclaveUUID: true,
+		},
+		Statuses: map[enclave.EnclaveStatus]bool{
+			enclave.EnclaveStatus_Running: true,
+		},
+	}
+
+	enclaves, err := pool.kurtosisBackend.GetEnclaves(ctx, filters)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting enclaves using filters '%+v'", filters)
+	}
+	enclavesLen := len(enclaves)
+
+	if enclavesLen == 0 {
+		return nil, stacktrace.NewError("There is not any running enclave with UUID '%v', it could have been stopped or destroyed", enclaveUUID)
+	}
+	if enclavesLen > 1 {
+		return nil, stacktrace.NewError("Expected to find only one running enclave with UUID '%v', but '%v' were found, this is a bug in Kurtosis", enclaveUUID, enclavesLen)
+	}
+
+	enclaveObj, found := enclaves[enclaveUUID]
+	if !found {
+		return nil, stacktrace.NewError("Expected to find an enclave with UUID '%v' in enclave map '%+v', but none was found; this is a bug in Kurtosis", enclaveUUID, enclaves)
+	}
+
+	return enclaveObj, nil
+}
+
+// The enclave pool feature is only available for Kubernetes so far, and it will be activated
+// only if users require this when setting the pool-size value
+func isEnclavePoolAllowedForThisConfig(
```

Comment:
Where is this used?

Reply:
@gbouv It's used in the `EnclaveManager` here:
<img width="865" alt="image" src="https://github.com/kurtosis-tech/kurtosis/assets/29951623/97cfcfac-380e-405e-90dd-360a2fb4864b">
but now we a second though I think that I'll move these validations inside the `CreateEnclavePool` method

Reply:
it was refactored as I described in the previous comment